libsplinter/src/rest_api/auth/roles/store/diesel/models.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::io::Write;
+
+use diesel::{
+    backend::Backend,
+    deserialize::{self, FromSql},
+    expression::{helper_types::AsExprOf, AsExpression},
+    serialize::{self, Output, ToSql},
+    sql_types::SmallInt,
+};
+
+use super::schema::{assignments, identities, role_permissions, roles};
+
+#[derive(Debug, PartialEq, Associations, Identifiable, Insertable, Queryable)]
+#[table_name = "roles"]
+#[primary_key(id)]
+pub struct RoleModel {
```

Comment:
The pub / pub(super) here should probably be swapped around, and other things in this module could be pub(super).